circuit/src/dynamic_sets/mod.rs

```diff
@@ -0,0 +1,490 @@
+						assigned_s.push(s)
+					}
+					// Move to the next row
+					ctx.next();
+				}
+
+				let mut assigned_ops = Vec::new();
+				for neighbour_ops in &self.ops {
+					let mut assigned_neighbour_op = Vec::new();
+					for chunk in neighbour_ops.chunks(ADVICE) {
+						for i in 0..chunk.len() {
+							let val = chunk[i].clone();
+							let s = ctx.assign_advice(config.common.advice[i], val)?;
+							assigned_neighbour_op.push(s)
+						}
+						// Move to the next row
+						ctx.next();
+					}
+					assigned_ops.push(assigned_neighbour_op);
+				}
+
+				let mut passed_s = Vec::new();
+				for i in 0..NUM_NEIGHBOURS {
+					let index = i % ADVICE;
+					let ps = ctx.assign_from_instance(
+						config.common.advice[index], config.common.instance, i,
+					)?;
+					passed_s.push(ps);
+					if i == ADVICE - 1 {
+						ctx.next();
+					}
+				}
+
+				let one = ctx.assign_from_constant(config.common.advice[0], Scalar::one())?;
+
+				let default_pk_x = ctx.assign_advice(
+					config.common.advice[1],
+					Value::known(PublicKey::default().0.x),
+				)?;
+
+				let default_pk_y = ctx.assign_advice(
+					config.common.advice[2],
+					Value::known(PublicKey::default().0.y),
+				)?;
+
+				Ok((
+					zero, assigned_pk_x, assigned_pk_y, assigned_big_r_x, assigned_big_r_y,
+					assigned_s, scale, assigned_ops, assigned_initial_score, assigned_total_score,
+					passed_s, one, default_pk_x, default_pk_y,
+				))
+			},
+		)?;
+
+		// signature verification
+		{
+			let mut pk_sponge = SpongeHasher::new();
+			pk_sponge.update(&pk_x);
+			pk_sponge.update(&pk_y);
+			let keys_message_hash = pk_sponge.synthesize(
+				&config.common,
+				&config.sponge,
+				layouter.namespace(|| "keys_sponge"),
+			)?;
+			for i in 0..NUM_NEIGHBOURS {
+				let mut scores_sponge = SpongeHasher::new();
+				scores_sponge.update(&ops[i]);
+				let scores_message_hash = scores_sponge.synthesize(
+					&config.common,
+					&config.sponge,
+					layouter.namespace(|| "scores_sponge"),
+				)?;
+				let message_hash_input = [
+					keys_message_hash.clone(),
+					scores_message_hash,
+					zero.clone(),
+					zero.clone(),
+					zero.clone(),
+				];
+				let poseidon = PoseidonHasher::new(message_hash_input);
+				let res = poseidon.synthesize(
+					&config.common,
+					&config.poseidon,
+					layouter.namespace(|| "message_hash"),
+				)?;
+
+				let eddsa = Eddsa::new(
+					big_r_x[i].clone(),
+					big_r_y[i].clone(),
+					s[i].clone(),
+					pk_x[i].clone(),
+					pk_y[i].clone(),
+					res[0].clone(),
+				);
+				eddsa.synthesize(
+					&config.common,
+					&config.eddsa,
+					layouter.namespace(|| "eddsa"),
+				)?;
+			}
+		}
+
+		// filter peers' ops
+		let ops = {
+			let mut filtered_ops = Vec::new();
+			for i in 0..NUM_NEIGHBOURS {
+				let mut ops_i = ops[i].clone();
+
+				// Update the opinion array - pairs of (key, score)
+				//
+				// TODO: How to nullify the op_score according to pair of (set_pk, op_pk)
```

Comment:
We select `0` or `org_score` based on some condition using `SelectChipset`.

For example, if the condition is unequal keys:
```rust
let is_equal_pk = IsEqaulChipset::new(set_pk, curr_pk);
let maybe_nullified_score = SelectChipset(is_equal_pk, org_score, zero);
// Mutate the opinion array at index `i` with `maybe_nullified_score`
```